### PR TITLE
[#124545911] Rebase our gpg feature onto upstream

### PR DESCRIPTION
**Pull request by benhyland**
## What

Following our Concourse upgrade we want to upgrade our various resources.

One of our resources is this git-resource fork, which we maintain only for the gpg feature.
We'll soon try to get that feature into upstream.

In the meantime, we rebase our feature onto upstream at the latest version contained in our Concourse version, which we believe is
https://github.com/concourse/git-resource/commit/72182ebe2157f64bcf85374e31c783293569f429

As part of this rebase, we also rework some commit messages and tidy the implementation in preparation for the upstream PR.

We are targeting a new branch, `new_gds_master`. This allows us to review and merge this PR without affecting production. Then, after the story is reviewed, we have a simple change to reset `gds_master` to the state of `new_gds_master` in order to deploy the change. We choose to do it this way to avoid the review process from having to deal with merge conflicts.

## How to Review

Locally, `docker build .` and verify that it completes successfully.
Update your pipeline by adding `tag: merge_with_upstream_take2` to the `source` part of the `git-gpg` resource type definition in `paas-cf/concourse/pipelines/create-bosh-cloudfoundry.yml`.
Using the updated pipeline, run a deploy in your personal environment and see that git resources are behaving normally.

Once you're happy the change should go out, merge this PR. Then update the `gds_master` branch to point at the tip of `new_gds_master`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,11 @@
 FROM alpine:3.4
 
-RUN apk add --update curl git openssh-client jq perl && rm -rf /var/cache/apk/*
+# Necessary to get newer version with fix for https://bugs.gnupg.org/gnupg/issue2371
+# FIXME: remove this once https://bugs.alpinelinux.org/issues/6112 has been resolved.
+RUN wget http://dl-4.alpinelinux.org/alpine/edge/main/x86_64/libgpg-error-1.24-r0.apk \
+  && apk add libgpg-error-1.24-r0.apk && rm libgpg-error-1.24-r0.apk
+
```
Comment (saliceti) [inline Dockerfile]: You can use apk to download from the edge repo:

```
RUN apk add --update libgpg-error=1.24-r0 --repository http://dl-cdn.alpinelinux.org/alpine/edge/main/
```

Comment (benhyland) [inline Dockerfile]: Done, thanks for the suggestion

**Comment by saliceti** [comment]
Tested successfully signed commits and tag filter.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,20 @@
-FROM concourse/buildroot:git
+FROM alpine:3.4
+
+# Necessary to get newer version with fix for https://bugs.gnupg.org/gnupg/issue2371
+# FIXME: remove this once https://bugs.alpinelinux.org/issues/6112 has been resolved.
+RUN apk add --update libgpg-error=1.24-r0 --repository http://dl-cdn.alpinelinux.org/alpine/edge/main/
```
Comment (alext) [inline Dockerfile]: If picking a particular repo, we can probably avoid pinning the version, which would make this more robust to changes in the edge repo.

Comment (saliceti) [inline Dockerfile]: I tried without the version but for some reason sometimes it would show both versions installed.

**Pull request merged**